File: app/src/main/java/space/pal/sig/view/facts/FactsViewModel.java
package space.pal.sig.view.facts;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

import space.pal.sig.model.Fact;
import space.pal.sig.repository.FactRepository;

/**
 * SpaceNow
 * Created by Catalin on 7/29/2020
 **/
public class FactsViewModel extends AndroidViewModel {

    private final LiveData<List<Fact>> facts;

    public FactsViewModel(@NonNull Application application, FactRepository factRepository) {
        super(application);
        facts = factRepository.findAll();
    }

    public LiveData<List<Fact>> getFacts() {
        return facts;
    }
}

File: app/src/main/java/space/pal/sig/view/facts/FactsViewModel.kt
package space.pal.sig.view.facts

import android.app.Application
import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData
import space.pal.sig.model.dto.FactDto
import space.pal.sig.repository.FactRepository
import space.pal.sig.view.BaseViewModel

/**
 * SpaceNow
 * Created by Catalin on 12/24/2020
 **/
class FactsViewModel(
        application: Application,
        factRepository: FactRepository
) : BaseViewModel(application) {

    private val facts = MediatorLiveData<List<FactDto>>()

    init {
        facts.addSource(factRepository.findAll()) { facts ->
            this.facts.value = facts.mapIndexed { index, fact ->
                FactDto(fact.content, "#${index + 1}")
            }.shuffled()
        }
    }

    fun getFacts(): LiveData<List<FactDto>> {
        return facts
    }

}